pkg/cloud/services/network/vpc.go

```diff
@@ -66,6 +66,20 @@ func (s *Service) reconcileVPC() error {
 			return nil
 		}
 
+		// Make sure tags are up-to-date.
+		// **Only** do this for managed VPCs. Make sure this logic is below the above `vpc.IsUnmanaged` check.
```

Comment:
Shall we not apply additionalTags to the unmanaged VPC? AdditionalTags are user defined tags that could be applied to managed/unmanaged VPC right?
Please don't mind if I don't have any prior context.

Reply:
Hmmm. I kind of want to avoid tagging unmanaged resources. This doesn't discriminate between user defined tags either. Because of the previous tags assignment. If any user adds a tag on the vpc itself this will nuke it and override the whole thing with what's set on the cluster and the additional tags. I think... :D but let me make sure about that assumption.

Reply:
Ok, so this actually worked since there is a describe in front of this where it gets the tag and then does a diff. I'll push and run some tests, hopefully nothing will break. :D

Reply:
Nooooo no no no this actually didn't work at all. It completely messed up the tags on the unmanaged VPC. :D So reverting to the previous one. :)

Reply:
But is there a way we can somehow do this for unmanaged VPC as well, as additionalTags can be applied to both managed/unmanaged VPC

Reply:
I don't see how. I don't think we should manage tags on an existing VPC. Additional Tags are no exception. If the user defines a VPC we shouldn't touch it even if there are additional tags defined. It just means trouble. And when I tried it it got into all sorts of problems when I started changing tags on the existing VPC that CAPA should not have managed.

Reply:
Hmm makes sense to me. 🙂 
We can see it later if some request comes regarding this.

Reply:
Sounds good! :)